### ISSUE TITLE
Getting  error : Method invocation failed because [System.Char] does not contain a method named 'StartsWith'

### ISSUE THREAD

**Issue by patelrk** [qetza/vsts-replacetokens-task#3]
Hi @qetza,

I get the following when the token replacement task runs through an azure worker role project.  It handled .config fine without these errors in the same project but bot the task for *.cscfg.

I've copied out part of the log for your convenience:

2016-03-11T08:09:09.2950174Z Starting task: Replace tokens in $(System.DefaultWorkingDirectory)$(PackagePath)\Workers\**\*.cscfg
 
2016-03-11T08:09:09.4197945Z Executing the powershell script: C:\BuildAgent\tasks\replacetokens\1.1.0\replace-tokens.ps1
 
2016-03-11T08:09:10.1073335Z ##[error]Method invocation failed because [System.Char] does not contain a method named 'StartsWith'.
 
[... There were a hundred odd errors exactly the same as the one above which i have left out for clarity]
 
 
2016-03-11T08:09:10.3887591Z Replacing tokens in file 'C:\BuildAgent\_work\35a32acd6\New.Product.CI\drop\Workers\bin\CI\app.publish\ServiceConfiguration.CloudCI.cscfg'...
 
2016-03-11T08:09:10.5917956Z Finishing task: replacetokens

**Comment by jessehouwing** [comment]
I suspect it's a bug in `Get-MatchingFiles` where `$MatchingFiles` is treated as a string instead of a `[string[]]` I've had the same issue with a couple of my other tasks and found that explcitly declarign the variable with:

```
[string[]] $MatchingFiles = @()
```

Helps. As does using the `@(Find-Files....)` and `@(Split...)` notation to always capure the output of these parameters as an array.

This probably works better:

```
Function Get-MatchingFiles
{
    [CmdletBinding()]
    param(
        [string] $Pattern,
        [string] $Root
    )
    
    [string[]]$matchingFiles = @()

    if ($Pattern.Contains("*") -or $Pattern.Contains("?"))
    {
        $matchingFiles += @(Find-Files -SearchPattern $Pattern -Root $Root)
        if (!$matchingFiles.Count)
        {
            throw "Target files not found using search pattern '${Pattern}'."
        }
    }
    elseif ($Pattern.Contains(';'))
    {
        $matchingFiles += @($Pattern -split ';')
    }
    else
    {
        $matchingFiles += @($Pattern)
    }

    for ($i = 0 ; $i -lt $matchingFiles.Length ; ++$i) {
        if (!$matchingFiles[$i].StartsWith($Root))
        {
            $matchingFiles[$i] = (Join-Path $Root $matchingFiles[$i])
        }
    }
    
    $matchingFiles
}
```

**Comment by qetza** [comment]
Hi @patelrk,

Could send the value you specify in `Target Files` to validate Jesse's assomption and fix?

Thanks

**Comment by jessehouwing** [comment]
Looking at what it expands to in the pasted logs it seems to be:
```
$(System.DefaultWorkingDirectory)$(PackagePath)\Workers\**\*.cscfg
```

**Comment by qetza** [comment]
@jessehouwing yeah powershell has a strange way to handle arrays and return values. I thought the code was okay because split should always return an array and i've added the comma to force an array but it seems that in @patelrk case the variable is an array of char... Just wanted to validate that the fix will force an array of string.

**Comment by patelrk** [comment]
Hi @qetza, i'm trying to validate @jessehouwing fix by running that function locally on my pc and its complaining that it doesn't recognise Find-Files.  Am i missing something?

**Comment by jessehouwing** [comment]
Find-files is specific to the build agent and doesn't work on a normal
powershell console. If you're running a local agent you could open the
agent\tasks folder and overwrite the powershell in that folder with the
patched version, that should not be overwritten until the next version
comes out.

**Comment by patelrk** [comment]
Hi @jessehouwing,

Thanks for the tip of how to test this, very useful!  Is simplified my input to Target Files to **\*.cscfg and that works with your fix. Thank you!

**Comment by qetza** [comment]
@patelrk i found the root cause of your first issue, when Find-Files only find one file the return value is automatically expanded by PowerShell as a single string. @jessehouwing fix will make sure the value is always an array.
On your second issue (multiple patterns), the semi-colon split is done by the Find-Files and should work. I'll do some testing today to check.

**Comment by qetza** [comment]
Fix in release 1.1.1
